### PR TITLE
RAC-461: add cleanup by blacklisted attribute codes

### PR DESCRIPTION
**Pull request by juliensnz**
**Description (for Contributor and Core Developer)**

Screenshot 2021-01-12 at 18 25 43

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

**Review by StevenVAIDIE — APPROVED** [review]

**Review by SamirBoulil — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Bundle/Command/CleanRemovedAttributesFromProductAndProductModelCommand.php [hunk 1]
```diff
@@ -195,6 +207,13 @@ private function launchCleanTask(array $productIds, string $env, string $rootDir
         $process->run();
     }
 
+    private function cleanBlacklistedAttributes(SymfonyStyle $io): void
+    {
+        $blacklistedAttributeCodes = $this->getAllBlacklistedAttributeCodes->execute();
```
Comment (ValentinMumble) [inline src/Akeneo/Pim/Enrichment/Bundle/Command/CleanRemovedAttributesFromProductAndProductModelCommand.php:212]: I remember we defined displaying the found blacklisted attributes as an acceptance criteria https://akeneo.atlassian.net/browse/RAC-461, do we still want this?
Also maybe we should not launch the job if no blacklisted attributes is found?

File: tests/back/Pim/Structure/Integration/Attribute/Query/GetAllBlacklistedAttributeCodesIntegration.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    public function test_it_returns_an_array_of_blacklisted_attributes(): void
+    {
+        $blacklister = $this->getBlacklister();
+        $blacklister->blacklist('description');
+        $blacklister->blacklist('name');
+
+        $query = $this->getQuery();
+        $result = $query->execute();
+
+        $this->assertEquals($result, ['description', 'name']);
```
Comment (ValentinMumble) [inline tests/back/Pim/Structure/Integration/Attribute/Query/GetAllBlacklistedAttributeCodesIntegration.php:33]:
```suggestion
        $this->assertEquals(['description', 'name'], $result);
```
Not important at all but it should be expected first

**Review by ValentinMumble — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Bundle/Command/CleanRemovedAttributesFromProductAndProductModelCommand.php (hunk 1 above)
Comment (juliensnz) [reply src/Akeneo/Pim/Enrichment/Bundle/Command/CleanRemovedAttributesFromProductAndProductModelCommand.php:212]: In the current state of the PR, it displays the list of attributes and does nothing if there is no blacklisted attributes

File: tests/back/Pim/Structure/Integration/Attribute/Query/GetAllBlacklistedAttributeCodesIntegration.php (hunk 2 above)
Comment (juliensnz) [reply tests/back/Pim/Structure/Integration/Attribute/Query/GetAllBlacklistedAttributeCodesIntegration.php:33]: ok, perfect, I will change that 👍

File: src/Akeneo/Pim/Enrichment/Bundle/Command/CleanRemovedAttributesFromProductAndProductModelCommand.php (hunk 1 above)
Comment (ValentinMumble) [reply src/Akeneo/Pim/Enrichment/Bundle/Command/CleanRemovedAttributesFromProductAndProductModelCommand.php:212]: AH yes I didn't see your screenshot but it still launches a job when empty I guess?

**Pull request merged**